### PR TITLE
feat (package/codemod): Add upgrade command to run codemod bundle.

### PR DESCRIPTION
**Pull request by shaper**

### CODE REVIEW

File: packages/codemod/src/lib/upgrade.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  const aiVersion = pkg.dependencies?.ai || pkg.devDependencies?.ai;
+
+  if (!aiVersion) {
+    throw new Error('No ai package found in dependencies');
+  }
+
+  const version = semver.clean(aiVersion.replace(/^[\^~]/, ''));
+  if (!version || !semver.gte(version, '3.4.0')) {
+    throw new Error('ai package must be at least version 3.4.0');
+  }
```
Comment (lgrammel) [inline packages/codemod/src/lib/upgrade.ts:32]: Not sure if we need this requirement? it could also work with e.g. 3.2 or 3.3

**Review by lgrammel — APPROVED** [review]

File: packages/codemod/src/lib/upgrade.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import fs from 'fs';
+import path from 'path';
+import semver from 'semver';
+import { transform } from './transform';
+import { TransformOptions } from './transform-options';
+
+const bundle = [
```
[the remark this answers is not in the document]
Comment (lgrammel) [reply packages/codemod/src/lib/upgrade.ts:7]: makes sense. could be as easy as `4.0/...`

File: packages/codemod/src/lib/upgrade.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  'remove-ai-stream-methods-from-stream-text-result',
+  'remove-experimental-ai-fn-exports',
+  'replace-baseurl',
+  'replace-nanoid',
+  'rewrite-framework-imports',
```
Comment (lgrammel) [inline packages/codemod/src/lib/upgrade.ts:12]: what does this particular migration do? (`rewrite-framework-imports`)

Comment (shaper) [reply packages/codemod/src/lib/upgrade.ts:12]: It rewrites for [this](https://sdk.vercel.ai/docs/troubleshooting/migration-guide/migration-guide-4-0#removed-svelte-vue-and-solidjs-exports), across any of svelte/vue/solid:

```
import { useChat } from 'ai/solid';
```

to

```
import { useChat } from '@ai-sdk/solid';
```

**Pull request merged**